File: ObjectRequestBrokerJava/src/main/java/vson/writers/ObjectWriter.java
package vson.writers;


import java.lang.reflect.Field;

import static vson.FieldUtils.getAllFields;
import static vson.FieldUtils.isPrimitive;

public class ObjectWriter {

    public ObjectWriter() {
    }


    public static String toJson(Object object) {
        StringBuilder sb = new StringBuilder();

        Class<?> mClass = object.getClass();

        sb.append("{\n").append("\"").append("type").append("\"").append(": "); // "type":"ClassName"
        sb.append("\"").append(mClass.getName()).append("\"").append(",\n");

        if (isPrimitive(mClass) || mClass.equals(String.class)){
            sb.append("\"value\": ").append(FieldWriter.writeFieldValue(object)).append("\n}");
            return sb.toString();
        }

        sb.append("\"").append("properties").append("\"").append(": "); // "properties":{ "attributeName":"attributeValue" }

        sb.append("{");
        sb.append("\n\"@hash\": ").append(object.hashCode()).append(", ");
        Field[] mClassFields = getAllFields(mClass);
        for (Field f : mClassFields) {
            sb.append("\n");
            if (!f.isAccessible()) { // set private fields accessible in order to obtain their value
                f.setAccessible(true);
                sb.append(FieldWriter.toJson(f, object));
                f.setAccessible(false);
            } else {
                sb.append(FieldWriter.toJson(f, object));
            }
            sb.append(",");
        }
        // remove trailing comma
        sb.replace(sb.length() - 1, sb.length(), "").append("\n}");


        sb.append("\n}");
        return sb.toString();
    }

}

File: ObjectRequestBrokerCS/ORB/vson/writers/ObjectWriter.cs
using System.Text;

namespace ORB.vson.writers
{

    public static class ObjectWriter
    {

        public static string ToJson(object @object)
        {
            var sb = new StringBuilder();

            var mClass = @object.GetType();

            sb.Append("{\n").Append("\"").Append("type").Append("\"").Append(": "); // "type":"ClassName"

            sb.Append("\"").Append(mClass.FullName).Append("\"").Append(",\n");

            if (FieldUtils.IsPrimitive(mClass) || mClass == typeof(string))
            {
                sb.Append("\"value\": ").Append(FieldWriter.WriteFieldValue(@object)).Append("\n}");
                return sb.ToString();
            }

            sb.Append("\"").Append("properties").Append("\"").Append(": "); // "properties":{ "attributeName":"attributeValue" }

            sb.Append("{");
            sb.Append("\n\"@hash\": ").Append(@object.GetHashCode()).Append(", ");
            var mClassFields = FieldUtils.GetAllFields(mClass);
            foreach (var f in mClassFields)
            {
                sb.Append("\n");


                sb.Append(FieldWriter.ToJson(f, @object));
                sb.Append(",");
            }
            // remove trailing comma
            sb.Remove(sb.Length - 1, 1).Append("\n}");


            sb.Append("\n}");
            return sb.ToString();
        }

    }

}
